subprojects/core/src/main/java/org/gradle/api/internal/tasks/cache/diagnostics/TaskOutputCacheDiagnostics.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.cache.diagnostics;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskOutputCacheDiagnostics {
+    List<TaskOutputCacheEvent> events = new ArrayList<TaskOutputCacheEvent>();
```

Comment:
The Javadoc on [ListenerManager#getBroadcaster](https://github.com/gradle/gradle/blob/14fa807164b09bdc642155f2b5c97546c0f648af/subprojects/messaging/src/main/java/org/gradle/internal/event/ListenerManager.java#L58-L58) states that `A listener is used by a single thread at a time.`. This would mean that we do not need to care about concurrency at all. This was also what I had in mind when implementing it.